DuckDuckGo/SettingsState.swift

```diff
@@ -39,6 +39,7 @@ struct SettingsState {
     struct Subscription: Codable {
         var canPurchase: Bool
         var isSignedIn: Bool
+        var hasSubscription: Bool
```

Comment:
Additional bool was required to discern if there is a subscription at all since `hasActiveSubscription` was not enough.